update data backup

XX-Yin opened the pull request:
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
Save data in the main thread. 

### What issues or discussions does this update address?
I speculate that the occasional crashes are related to the conflicting interaction between the different threads (between backup saving and `_GenerateATrial`).  Here, I moved the backup to after the `GeneratedTrials._GenerateATrial(self.Channel4)` to avoid simultaneous running of these two functions.

https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/605#issuecomment-2342288067

### Describe the expected change in behavior from the perspective of the experimenter
No

### Describe any manual update steps for task computers
No

### Was this update tested in 446/447?
Tested on my own computer.

alexpiet commented:
But if saving takes .5s, that means the GUI will be unresponsive for .5s after every trial. This is not ideal

XX-Yin commented:
> But if saving takes .5s, that means the GUI will be unresponsive for .5s after every trial. This is not ideal

You are right. I updated it to still use a separate thread. And moved the backup to after the `GeneratedTrials._GenerateATrial(self.Channel4) ` as I found the conflict is mainly caused by interaction with `_GenerateATrial`.

XX-Yin commented:
@alexpiet Do you have extra concerns about this PR?

alexpiet commented on src/foraging_gui/Foraging.py:
```diff
@@ -3963,6 +3959,23 @@ def _StartTrialLoop(self,GeneratedTrials,worker1,worker_save):
                 if self.NewTrialRewardOrder==1:
                     GeneratedTrials._GenerateATrial(self.Channel4)   
+                '''
+                ### Save data in the main thread. Keep it temporarily for testing ### 
```
Did you mean to remove this part, saving data in the main thread?

XX-Yin replied:
Yes, I discarded this the strategy to 'save data in the main thread'. Instead, I moved the previous `saving code` to the after of `GeneratedTrials._GenerateATrial(self.Channel4)` to make sure they are not running simultaneously.

alexpiet replied:
I'm confused. Line 3967 shows saving the data in the main thread, then line 3975 shows saving in a separate thread. Did you intend to remove the saving in the main thread?

XX-Yin replied:
The code for "saving data in the main thread" has been commented out with ''' '''

alexpiet replied:
Oh duh. I didn't see the multiline comment ```, I was looking for a # before each line
